kommandir/bin/adept_openstack.py

```diff
@@ -1357,18 +1356,17 @@ def activate_and_setup(namespace, venvdir, requirements, onlybin, nobin):
 # N/B: Any/All names used here are in the global scope
 if __name__ == '__main__':  # pylint: disable=C0103
     # Parse arguments
-    basename = os.path.basename(sys.argv[0])
-    if basename == DISCOVER_CREATE_NAME:
-        _dargs = parse_args(sys.argv, 'discover')
-    elif basename == ONLY_CREATE_NAME:
-        _dargs = parse_args(sys.argv, 'exclusive')
-    elif basename == DESTROY_NAME:
-        _dargs = parse_args(sys.argv, 'destroy')
-    elif basename == REAP_NAME:
-        _dargs = parse_args(sys.argv, 'reap')
-    else:
-        logging.error("Script was not called as %s, %s, or %s", *ALLOWED_NAMES)
+    basename = os.path.basename(sys.argv[0]).rstrip('.py')
```

Comment:
somehow(?) this rstrip seems to be eating more than just the '.py' :confused:

Reply:
my gosh that's odd...is ``rstrip()`` broken/bugged?

Reply:
Yeah, python strip() doesn't seem to work the way I thought it did. Looking.

Reply:
Oooohhhhh.....it's ``rstrip([CHARS])`` as in, characters to remove, not string.

Reply:
This should work:

```python
    basename = os.path.basename(sys.argv[0]).rpartition('.py')[0]
    op_requested = basename.partition('openstack_')[-1]
```

Reply:
New force push. Using `re`: `{,r}partition` seem close but could fail in (unlikely) corner cases. Might as well do it right. Tests pass.